src/Database/Concerns/HasRelationships.php

```diff
@@ -309,18 +324,21 @@ protected function handleRelation($relationName)
         switch ($relationType) {
             case 'hasOne':
             case 'hasMany':
-                $relation = $this->validateRelationArgs($relationName, ['key', 'otherKey']);
-                $relationObj = $this->$relationType($relation[0], $relation['key'], $relation['otherKey'], $relationName);
+                $relation = $this->validateRelationArgs($relationName, ['key', 'relatedKey']);
+                $relationObj = $this->$relationType($relation[0], $relation['key'], $relation['relatedKey'], $relationName);
                 break;
 
             case 'belongsTo':
-                $relation = $this->validateRelationArgs($relationName, ['key', 'otherKey']);
-                $relationObj = $this->$relationType($relation[0], $relation['key'], $relation['otherKey'], $relationName);
+                $relation = $this->validateRelationArgs($relationName, ['key', 'relatedKey']);
+                $relationObj = $this->$relationType($relation[0], $relation['key'], $relation['relatedKey'], $relationName);
                 break;
 
             case 'belongsToMany':
-                $relation = $this->validateRelationArgs($relationName, ['table', 'key', 'otherKey', 'parentKey', 'relatedKey', 'pivot', 'timestamps']);
-                $relationObj = $this->$relationType($relation[0], $relation['table'], $relation['key'], $relation['otherKey'], $relation['parentKey'], $relation['relatedKey'], $relationName);
+                $relation = $this->validateRelationArgs($relationName, ['table', 'key', 'relatedPivotKey', 'parentKey', 'relatedKey', 'pivot', 'timestamps']);
+                $relationObj = $this->$relationType($relation[0], $relation['table'], $relation['key'], $relation['relatedKey'], $relation['parentKey'], $relation['relatedKey'], $relationName);
+                if (!empty($relation['pivotModel'])) {
+                    $relationObj->using($relation['pivotModel']);
+                }
```

Comment:
This needs to be added to the other pivot model using relations (morphToMany & morphedByMany)